src/app/feedback-form/feedback-form.component.ts

```diff
@@ -1,16 +1,34 @@
 import { Component, Input } from '@angular/core';
+import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
 
 @Component({
   selector: 'ngt-feedback-form',
   standalone: true,
-  imports: [],
+  imports: [ReactiveFormsModule],
   templateUrl: './feedback-form.component.html',
-  styleUrl: './feedback-form.component.scss'
+  styleUrl: './feedback-form.component.scss',
 })
 export class FeedbackFormComponent {
   @Input() feedback: string = '';
 
   changeText(newText: string) {
     this.feedback = newText;
   }
+
+  form: FormGroup;
+
+  constructor(private readonly formBuilder: FormBuilder) {
+    this.form = this.formBuilder.group({
+      name: ['', Validators.required],
+      email: ['', [Validators.required, Validators.email]],
+      subject: ['', Validators.required],
+      message: ['', Validators.required],
+    });
+  }
+
+  onSubmit() {
+    if (this.form.valid) {
```

Comment:
this is nice but redundant. The button will be disabled if the form is invalid so you cant click it.